lib/utils/notification_service.dart

```diff
@@ -130,43 +138,204 @@ class NotificationService {
+        );
+        notificationDates.add(OSDDateTime(
+          year: dateTime.year,
+          month: dateTime.month,
+          day: dateTime.day,
+          hour: dateTime.hour,
+          minute: dateTime.minute,
+          notificationId: notificationId,
+        ));
+        notificationId = getNotificationId();
+      }
 
-    /// Schedule notification
+      // increase day
+      dateTime = dateTime.add(const Duration(days: 1));
+    }
+
+    // saving notification dates in db
+    Utils.saveDateTimes(notificationDates);
+    Utils.logInfo(
+      '[NOTIFICATIONS] - Notifications were rescheduled',
+    );
+  }
+
+  void cancelTodayNotification() async {
+    final List<OSDDateTime> notificationDates = Utils.getDateTimes();
+
+    // cancel notification
+    if (notificationDates.isNotEmpty) {
+      final first = notificationDates.first;
+      await _flutterLocalNotificationsPlugin.cancel(first.notificationId);
+      notificationDates.remove(first);
+      Utils.saveDateTimes(notificationDates);
+    }
+
+    Utils.logInfo(
+      '[NOTIFICATIONS] - Notification was canceled for today',
+    );
+  }
+
+  void scheduleTodayNotification() async {
+    final notificationDates = Utils.getDateTimes();
+    final int notificationId = getNotificationId();
+
+    final TimeOfDay scheduleTime = getScheduledTime();
+    final DateTime todayDate = DateTime.now();
+    final dateTime = DateTime(
+      todayDate.year,
+      todayDate.month,
+      todayDate.day,
+      scheduleTime.hour,
+      scheduleTime.minute,
+    );
+
+    if (dateTime.isBefore(DateTime.now())) return;
+
+    // schedule notification
     await _flutterLocalNotificationsPlugin.zonedSchedule(
-      _notificationId,
+      notificationId,
       'notificationTitle'.tr,
       'notificationBody'.tr,
-      tz.TZDateTime.parse(tz.local, setTime),
+      tz.TZDateTime.parse(tz.local, dateTime.toString()),
       _platformNotificationDetails,
       uiLocalNotificationDateInterpretation:
-      UILocalNotificationDateInterpretation.absoluteTime,
+          UILocalNotificationDateInterpretation.absoluteTime,
       androidScheduleMode: AndroidScheduleMode.exactAllowWhileIdle,
-      // Allow notification to be shown daily
-      matchDateTimeComponents: DateTimeComponents.time,
+    );
+    notificationDates.insert(0, OSDDateTime(
+      year: dateTime.year,
+      month: dateTime.month,
+      day: dateTime.day,
+      hour: dateTime.hour,
+      minute: dateTime.minute,
+      notificationId: notificationId,
+    ));
+
+    // saving notification dates in db
+    Utils.saveDateTimes(notificationDates);
+    Utils.logInfo(
+      '[NOTIFICATIONS] - Notification was scheduled for today',
     );
   }
 
-  Future<void> rescheduleNotification(DateTime day) async {
-    final TimeOfDay timeOfDay = getScheduledTime();
-    await scheduleNotification(timeOfDay.hour, timeOfDay.minute, day);
+  int getNotificationId() {
```

Comment:
Is there any reason to have a random id?

Reply:
We are scheduling 2 weeks notifications and storing this in shared pref and canceling notification based on notification id. So for unique identification using random id.